python-threatexchange/threatexchange/extensions/vpdq/vpdq_index.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Implementation of SignalTypeIndex abstraction for PDQ by wrapping
+hashing.pdq_faiss_matcher.
+"""
+
+import typing as t
+import vpdq
+from threatexchange.signal_type.index import (
+    PickledSignalTypeIndex,
+    VPDQIndexMatch,
+    IndexMatch,
+    T as IndexT,
+)
+from threatexchange.extensions.vpdq.vpdq_faiss import VPDQHashIndex
+from threatexchange.extensions.vpdq.vpdq_util import (
+    prepare_vpdq_feature,
+    VPDQ_QUALITY_THRESHOLD,
+    VPDQ_DISTANCE_THRESHOLD,
+)
+
+
+class VPDQIndex(PickledSignalTypeIndex[IndexT]):
+    """
+    Wrapper around the vpdq faiss index lib using VPDQHashIndex
+    """
+
+    def __init__(
+        self,
+        quality_threshold: int = VPDQ_QUALITY_THRESHOLD,
+    ) -> None:
+        super().__init__()
+        self.index: VPDQHashIndex = VPDQHashIndex()
+        self._entry_idx_to_features_and_entires: t.List[
+            t.Tuple[t.List[vpdq.VpdqFeature], IndexT]
+        ] = []
+        self._index_idx_to_vpdqHex_and_entry: t.List[t.Tuple[int, t.List[int]]] = []
+        self._unique_vpdqHex_to_idx: t.Dict[str, int] = {}
```

Comment:
Which idx? We have index_idx and entry_idx (assume entry)

Reply:
It's actually  _unique_vpdqHex_to_index_idx, because each unique hex hash should only map to one index_idx in faiss.